File: src/main/java/com/vokerg/blog/VokergBlog/model/Article.java
package com.vokerg.blog.VokergBlog.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document(collection="articles")
@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Article {
    @Id
    private String id;
    private String title;
    private String subject;
    private String content;
    private String idAuthor;
    private String idSharedArticle;
    private String author;
    private List<String> tags;
}

File: client/src/app/modules/model/article.ts
export class Article {
  id: string;
  title: string;
  subject: string;
  content: string;
  idAuthor: string;
  idSharedArticle: string;
  author: string;
  liked: boolean;
  tags: string[];
  likeCount: number;
  commentsCount: number;
  sharedArticle: Article
  constructor() {
    this.id = null;
    this.title = "";
    this.subject = "";
    this.content = "";
    this.author = "";
    this.liked = false;
    this.tags = [];
    this.likeCount = 0;
    this.commentsCount = 0;
  }
}
